crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -393,8 +394,7 @@ impl C8YHttpProxyActor {
                 end_point.get_url_for_event_binary_upload(&event_response_id);
             Ok(HttpRequestBuilder::post(&binary_upload_event_url)
                 .header("Accept", "application/json")
-                .header("Content-Type", "text/plain")
-                .body(file_content.to_string()))
+                .body(file_content.clone()))
```

Comment:
It's unfortunate that we have to clone the whole file contents here again. But, I see how the closure forced you into it. Could we have used an `Arc` over that `file_content` Vector to avoid this?

Reply:
It is unfortunate, but no different to what was here previously, just a bit more explicit.

Reply:
How about reading the file content from inside the closure? I quickly tried it and it seemed to work. The only thing was that I had to switch back to `std::fs::read` from `tokio::fs::read`. Do you see any problem with that?

Reply:
I know we were using std::fs::read, but blocking in an async context in tokio is usually a really bad idea. Because thin edge spawns a bunch of small tasks (and therefore tokio runs them on multiple OS threads), it's not that likely to cause an issue, but on thick edge, I've seen issues where async functions completely fail to make any progress, due in part to executing a blocking operation (in that case it was println! that was the culprit)

Reply:
That's not to say we can't mitigate this either with spawn_blocking, or by returning a future from the closure, just that naïvely using blocking functions to read files is something that is actually worth avoiding

Reply:
Even without the `clone()` problem, we are reading the entire file content to memory. Is there any way to avoid that. I have seen the `reqwest` crate accepting a `File` handle directly into its body. Can we do something similar here? Not necessarily in this PR, if not trivial. Otherwise let's just defer it to a follow-up PR. No point blocking this PR for it.